FWCore/Framework/src/Event.cc

```diff
@@ -52,7 +52,9 @@ namespace edm {
   }
 
   void
-  Event::setProducer( ProducerBase const* iProd, std::vector<BranchID>* previousParentage) {
+  Event::setProducer(ProducerBase const* iProd,
+                     std::vector<BranchID>* previousParentage,
+                     std::vector<BranchID>* gotBranchIDsFromAcquire) {
```

Comment:
Let's break this into two functions, the original `setProducer` and a new `setProducerForAcquire`. Essentially the `if(previousParentage)` is already trying to make this one method do the function of two methods.

Reply:
Done, I split setProducer into 2 functions.